products.js

```diff
@@ -197,44 +197,242 @@ function tunedTitleFnodes(nodeToCssMap) {
     return tuningRoutine;
 }
 
-function tunedPriceFnodes(nodeToCssMap, coeffDollarSign = 1, coeffCSS = 1, coeffSavings = 0.5) {
+
+function tunedPriceFnodes(nodeToCssMap, coeffDollarSign = 1.6, coeffNearDollarSign = 1, coeffHasNumbers = 0.7,
+  coeffSpanBonus = 0.7, coeffSemanticTags = 1.9000000000000001, coeffCurrentPrice = 0.39999999999999997, coeffItemprop = 1, coeffKeywords = 1.3,
+  coeffStrike = 0.7, coeffNotSavings = 1.3, coeffAboveFold = 1.3, coeffCenterRight = 1.3, coeffMiddleHeight = 0.09999999999999998,
+  coeffBolded = 1, coeffNumNumbers = 0.39999999999999997, coeffNumDollarSigns = 0.39999999999999997, coeffPriceFormat = 0.7, coeffNumDots = 1.9000000000000001, coeffMetaTag = 7.3) {
 
     function hasDollarSign(fnode){
-      if(fnode.element.childNodes[0] && fnode.element.childNodes[0].nodeValue && fnode.element.childNodes[0].nodeValue.includes('$')){
+      if(fnode.element.textContent.includes('$')){
         return 2 * coeffDollarSign;
       }
       return 1;
     }
 
+    function nearDollarSign(fnode){
+      if(fnode.element.previousSibling && fnode.element.previousSibling.textContent.includes('$')){
+        return 3 * coeffNearDollarSign;
+      }
+      return 1;
+    }
+
+    function hasNumbers(fnode){
+      const regex = new RegExp(".*[0-9].*");
+      if(fnode.element.textContent.match(regex)){
+        return 10 * coeffHasNumbers;
+      }
+      return 1;
+    }
+
+
+    function spanBonus(fnode){
+      if(fnode.element.tagName === 'SPAN'){
+        return 2 * coeffSpanBonus;
+      }
+      return 1;
+    }
+
+    function semanticTags(fnode){
+      if(fnode.element.getElementsByTagName('SUP').length > 0 || fnode.element.getElementsByTagName('STRONG') > 0){
+        return 2 * coeffSemanticTags;
+      }
+      return 1;
+    }
+
+    function priceIsCurrent(fnode){
+      if(fnode.element.id.match(/(current|now)/i)){
```

Comment:
Another spot that will face problems on non-English pages. That's another direction you could attack.